homeassistant/components/http/ban.py

```diff
@@ -61,9 +62,19 @@ def setup_bans(hass, app, login_threshold):
 
     # Verify if IP is not banned
     ip_address_ = request[KEY_REAL_IP]
-    is_banned = any(
-        ip_ban.ip_address == ip_address_ for ip_ban in request.app[KEY_BANNED_IPS]
-    )
+
+    # Supervisor IP should never be banned
+    if "SUPERVISOR" in os.environ:
```

Comment:
It feels a bit weird that we check for the supervisor (hassio) integration specific environment key in the http integration. I didn't think of another solution yet.

Reply:
Can we trust that nothing else could set this environment key? Do we need to validate the supervisor ip some other way too?

Reply:
Can we check if the supervisor (hassio) integration is active/loaded?

In general, this is considered quite safe.

Reply:
let's add a helper function to hassio integration and do as Frenck suggests:

```python
if "hassio" in hass.config.components and hass.components.hassio.supervisor_ip() == …:
```